### PR TITLE
Saagar/itemtests

### PR DESCRIPTION
**Pull request by Saagar3690**
Finished tests for Item. 
Added the following tests:
1. POST items/add
   - Valid request
   - Missing image
   - Missing name and/or price and/or eventId in request
2. PUT items/weight
   - valid request
   - invalid itemId
   - missing itemId and/or weight in request
3. DELETE items
   - valid request
   - missing itemId and/or eventId
   - invalid itemId
   - invalid eventId

Screen Shot 2020-12-06 at 5 34 15 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,10 +8,10 @@
     "lint:fix": "eslint . --fix --ext .js",
     "dev": "export NODE_ENV=dev && npx nodemon server.js",
     "dev:win": "set NODE_ENV=dev && npx nodemon server.js",
-    "test": "export NODE_ENV=test && nyc --reporter=text-summary mocha --timeout 10000 --exit",
-    "test:win": "set NODE_ENV=test && nyc --reporter=text-summary mocha --timeout 10000 --exit",
-    "test:enforce": "export NODE_ENV=test && nyc --check-coverage --reporter=text mocha --timeout 10000 --exit",
-    "test:enforce:win": "set NODE_ENV=test && nyc --check-coverage --reporter=text mocha --timeout 10000 --exit"
+    "test": "export NODE_ENV=test && mocha test/bots.js --timeout 5000 --exit && mocha test/items.js --timeout 5000 --exit",
+    "test:win": "set NODE_ENV=test && mocha test/bots.js --exit && mocha test/items.js --exit",
```
Comment (dvorjackz) [inline package.json:12]: Add --timeout 5000's to the test:win too

File: test/items.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+			return await chai
+				.request(app)
+				.put('/items/weight')
+				.send({
+					itemId: 'clearlyWrongId',
+					weight: 12.7,
+				})
+				.then(async (res) => {
+					assert.notStrictEqual(res.status, 200);
```
Comment (dvorjackz) [inline test/items.js:148]: I think all resource not found statuses are 404, so you could say `strictEqual(res.status, 404)` here

File: test/items.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+			return await chai
+				.request(app)
+				.delete('/items')
+				.query({
+					itemId: 'clearlyWrongId',
+					eventId: event._id.toString(),
+				})
+				.then(async (res) => {
+					assert.notStrictEqual(res.status, 200);
```
Comment (dvorjackz) [inline test/items.js:200]: 404

File: test/items.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+			return await chai
+				.request(app)
+				.delete('/items')
+				.query({
+					itemId: savedItem._id.toString(),
+					eventId: 'clearlyWrongId',
+				})
+				.then(async (res) => {
+					assert.notStrictEqual(res.status, 200);
```
Comment (dvorjackz) [inline test/items.js:217]: 404

**Review by dvorjackz — CHANGES_REQUESTED** [review]
Seems to be your test event and test item leftover. Also, some linting errors, make sure to npm run lint and fix those

Screen Shot 2020-12-06 at 7 54 39 PM

**Review by Saagar3690 — COMMENTED** [review]
Made the changes as requested. Reverted the previous commit and undid the git add . I had done before. Minor changes to the item tests to ensure all items and events were deleted from test database.

**Comment by dvorjackz** [comment]
LGTM

**Review by dvorjackz — APPROVED** [review]

**Pull request merged**